lib/logstash/codecs/avro.rb

```diff
@@ -71,6 +74,13 @@ def decode(data)
     decoder = Avro::IO::BinaryDecoder.new(datum)
     datum_reader = Avro::IO::DatumReader.new(@schema)
     yield LogStash::Event.new(datum_reader.read(decoder))
+  rescue => e
+    if tag_on_failure
+      @logger.error("Avro parse error, original data now in message field", :error => e)
+      yield LogStash::Event.new("message" => data, "tags" => ["_avroparsefailure"])
+    else
+      raise e
+    end
   end
```

Comment:
Can we catch a more specific Exception?

I believe this codec is similar to what the json codec does which is https://github.com/logstash-plugins/logstash-codec-json/blob/master/lib/logstash/codecs/json.rb#L49 which is always `tag_on_failure`.

But this will make the change `somewhat` non backward compatible.